calm_adapter/window_generator/window_generator.py

```diff
@@ -0,0 +1,71 @@
+#!/bin/env python3
+
+from datetime import date, timedelta
+import click
+import json
+import boto3
+from tqdm import tqdm
+
+
+class WindowGenerator:
+
+    topic_arn = "arn:aws:sns:eu-west-1:760097843905:calm-adapter"
+    role_arn = "arn:aws:iam::760097843905:role/platform-developer"
+
+    def __init__(self, start, end=None):
+        self.start = start
+        self.end = end or start
+        assert self.start <= self.end, "Start window not be after end window"
+        self.client = self.sns_client
+
+    def run(self):
+        for window in tqdm([*self.windows]):
+            self.send_window_to_sns(window)
+
+    @property
+    def windows(self):
+        while self.start <= self.end:
+            yield self.start
+            self.start += timedelta(days=1)
+
+    def send_window_to_sns(self, window):
+        msg = json.dumps({"date": window.isoformat()})
+        self.client.publish(
+            TopicArn=self.topic_arn, Message=msg, Subject=f"Window sent by {__file__}"
+        )
+
+    @property
```

Comment:
minor: the general trend is to reserve `@property` for methods that:

* Return immediately
* Will never throw an exception

So they might as well be a static attribute. We shouldn't use `@property` here, because neither of those rules hold.

```
self.sns_client  # we expect won't throw, returns immediately

self.sns_client()  # is doing something interesting, might delay, might throw
```

Reply:
Hmm I'm not sure that is always the trend, especially as this is python and I would expect things to throw at any time. Where I have coded python previously we agreed that things which have clear side effects should be called  as a function, but properties are fine elsewhere.

Happy to change here tho to be more consistent with the rest of the code base.

Reply:
(worth mentioning this does obviously have a side effect of calling an external API, but does not really impact on the application beyond exception)